syntax = "proto3";

package gfs;

service MasterServerToClient {
    rpc ListFiles(String) returns (String) {}
    rpc CreateFile(String) returns (String) {}
    rpc AppendFile(String) returns (String) {}
    rpc CreateChunk(String) returns (String) {}
    rpc ReadFile(String) returns (String) {}
    rpc WriteFile(String) returns (String) {}
}

service ChunkServerToClient {
    rpc Create(String) returns (String) {}
    rpc GetChunkSpace(String) returns (String) {}
    rpc Append(String) returns (String) {}
    rpc Read(String) returns (String) {} 
    rpc AddData(String) returns (String) {}
    rpc Write(String) returns (String) {}
    
}

message String {
    string st = 1;
}

message Empty {

}

service Health {
    rpc Check(HealthCheckRequest) returns (HealthCheckResponse) {}
}

message HealthCheckRequest {
    string lease = 1;
}

message HealthCheckResponse {

    string status = 1;
}


service PrimaryToClient{
    rpc Commit (String) returns (String) {}
}

//  not used
message PrimaryToClientRequest {
    string clientid = 1;
}